src/game_inventory.cpp

```diff
@@ -90,26 +90,52 @@ void game_menus::inv::common( player &p )
 {
     static const std::set<int> allowed_selections = { { '\0', '=' } };
 
-    p.inv.restack( p );
+    bool keep_open = false;
+    item_location last_loc = item_location();
 
-    inventory_pick_selector inv_s( p );
+    // Outer loop for when we want to recreate the selector (something other than invlets changed)
+    do {
+        p.inv.restack( p );
 
-    inv_s.add_character_items( p );
-    inv_s.set_title( _( "Inventory" ) );
+        inventory_pick_selector inv_s( p );
 
-    int res;
-    do {
-        inv_s.set_hint( string_format(
-                            _( "Item hotkeys assigned: <color_light_gray>%d</color>/<color_light_gray>%d</color>" ),
-                            p.allocated_invlets().size(), inv_chars.size() ) );
-        const item_location &location = inv_s.execute();
-        if( location == item_location::nowhere ) {
-            break;
-        }
-        g->refresh_all();
-        res = g->inventory_item_menu( p.get_item_position( location.get_item() ) );
-        g->refresh_all();
-    } while( allowed_selections.count( res ) != 0 );
+        inv_s.add_character_items( p );
+        inv_s.set_title( _( "Inventory" ) );
+
+        int res = 0;
+        do {
+            inv_s.set_hint( string_format(
+                                _( "Item hotkeys assigned: <color_light_gray>%d</color>/<color_light_gray>%d</color>" ),
+                                p.allocated_invlets().size(), inv_chars.size() ) );
+
+            if( last_loc != item_location::nowhere ) {
```

Comment:
This looks like a hack.

Reply:
How so? Do you specifically mean `last_loc`?

If you're referring to the outer loop, it's kind of necessary, wherever you put it, and this seemed like the best place.

In general this could be improved by giving the ability to inventory_selectors to be cleared before filled again (which requires to pass again all items from player, map and vehicle anyway) and to keep some of the informations (perhaps a better way to remember location, filters and whatever else we might want later), but this is more likely to results in weird buggy behaviour as we change the entries from underneath the object. For now I decided to just recreate the selector, because this is likely to be necessary in some cases anyway (Like in #29148).

Extending the inventory_selector to be more dynamic is the next thing I'll do, but I just wanted to get a functional PR out beforehand so I have time to properly test some deeper changes, to get feedback and to allow my other PR which depends on it to finally be merged 😋